src/metabase/models/dashboard_card.clj

```diff
@@ -217,6 +221,117 @@
       (db/delete! DashboardCard :id (:id dashboard-card)))
     (events/publish-event! :dashboard-remove-cards {:id id :actor_id user-id :dashcards [dashboard-card]})))
 
+;;; ----------------------------------------------- Link cards ----------------------------------------------------
+
+(def ^:private all-card-info-columns
+  {:model         :text
+   :id            :integer
+   :name          :text
+   :description   :text
+
+   ;; for cards and datasets
+   :collection_id :integer
+   :display       :text
+
+   ;; for tables
+   :db_id        :integer})
+
+(def ^:private  link-card-columns-for-model*
+  {:database  [:id :name :description]
+   :table     [:id [:display_name :name] :description :db_id]
+   :dashboard [:id :name :description :collection_id]
+   :card      [:id :name :description :collection_id :display]
+   :dataset   [:id :name :description :collection_id :display]})
+
+(defn- ->column-alias
+  "Returns the column name. If the column is aliased, i.e. [`:original_name` `:aliased_name`], return the aliased
+  column name"
+  [column-or-aliased]
+  (if (sequential? column-or-aliased)
+    (second column-or-aliased)
+    column-or-aliased))
+
+(defn- link-card-columns-for-model
```

Comment:
I would call this `select-clause-for-model` because it exists to fill the gaps for the select query. `link-card-columns-for-model` isn't narrow enough. It would be nice if we could document this function a bit too. I like how `select-clause-for-model` is commented + spec'd, and I think this is complex enough that it should be commented + spec'd too. https://github.com/metabase/metabase/blob/d14d2a022a267c74fdbdf713f03d9cc3623f7428/src/metabase/api/search.clj#L138

Reply:
added!